### DISCUSSION TITLE
Unwanted vertical lines appear

### DISCUSSION THREAD

**Discussion by yzt2024** [jsvine/pdfplumber#1225] [Ask for help with specific PDFs]
Hello everyone,
I'm trying to export the table from the following pdf, but there are vertical lines that doesn't exist in the original text when the table is divided. I tried to modify snap_tolerance but did not get the expected division result. My parameters are as follows:

 table_setting={
"vertical_strategy": "lines",
"horizontal_strategy": "lines",
"snap_tolerance": 3,
"snap_tolerance": 3,
"join_tolerance": 3,
"edge_min_length": 3,
"min_words_vertical": 20,
"min_words_horizontal": 3,
"text_tolerance": 3,
"intersection_tolerance": 1,
}

[new.pdf](https://github.com/user-attachments/files/17807994/new.pdf)

![testpage134](https://github.com/user-attachments/assets/ac6375e6-5974-43ab-93f4-baa727ccef50)

if i use a snap_tolerance=4,then the division of the left boundary will be problematic.
![testpage134](https://github.com/user-attachments/assets/877f69bf-642d-49c3-90cb-a590478b668d)

So what am I supposed to do? I see from the discussion that boundaries can be determined manually by use the "explicit_vertical_lines", I haven't tried yet. I want to know if this parameter needs to specify all vertical lines,  because I actually only need to modify the leftmost boundary, can I modify only one line and the other vertical lines are still automatically generated? What should I do? Thank you very much!

**Comment by jsvine** [comment]
I would suggest examining the vertical lines / rects (`page.lines`, `page.rects`) to try to determine what properties are unique to the objects you want to exclude, and then using `filtered = page.filter(fn)` to create a filtered version of the page before extracting. There are a few other discussions that show examples of this, including this recent one: https://github.com/jsvine/pdfplumber/discussions/1219#discussioncomment-11210228

**Reply by yzt2024** [reply]
Thank you very much! I'm checking all the rects in the picture,I found  that the first cell of all tables contains three rects:
![test_line_rectold2134](https://github.com/user-attachments/assets/a0c37b29-85ed-4f9d-9b5e-5dda2983be19)
then i use the following code to delete the first rects and extend the width of the second rect.
![image](https://github.com/user-attachments/assets/9528cfdc-55fc-4565-9ea4-7b9e98580c52)
Now the division of the first row is correct, but the first column of the other rows still has the problem, I'll try to fix them
![testpage_change134](https://github.com/user-attachments/assets/b5408e25-a322-4f01-91e0-b8da858135b0)
